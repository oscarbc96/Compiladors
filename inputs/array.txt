calc on

[]
[1]

a := 9
b := true and false
c := [a, 4, 6.0, true, "afds", b]
d := []

c[0]
c[1]
c[2]
