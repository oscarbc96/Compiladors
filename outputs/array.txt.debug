FLEX: Detected mode CALC
BISON: Starting in mode CALC
FLEX: NEWLINE
FLEX: NEWLINE
FLEX: SQUARE_BRACKET_OPEN yytex: [
FLEX: SQUARE_BRACKET_CLOSE yytex: ]
FLEX: NEWLINE
BISON: EXPRESSION: TYPE: ARRAY VALUE: EMPTY
FLEX: SQUARE_BRACKET_OPEN yytex: [
FLEX: INT yytex: 1 -> 1
FLEX: SQUARE_BRACKET_CLOSE yytex: ]
FLEX: NEWLINE
BISON: EXPRESSION: TYPE: ARRAY VALUE: 1
FLEX: NEWLINE
FLEX: ASSIGN yytex: :=
FLEX: INT yytex: 9 -> 9
FLEX: NEWLINE
BISON: ASSIGN ID: a TYPE: INT VALUE: 9
FLEX: ASSIGN yytex: :=
FLEX: TTRUE yytex: true
FLEX: AND yytex: and
FLEX: TFALSE yytex: false
FLEX: NEWLINE
BISON: Performing and operation
BISON: ASSIGN ID: b TYPE: BOOL VALUE: false
FLEX: ASSIGN yytex: :=
FLEX: SQUARE_BRACKET_OPEN yytex: [
FLEX: ID yytex: a -> a
BISON: Looking for ID a
FLEX: COMMA yytex: ,
FLEX: INT yytex: 4 -> 4
FLEX: COMMA yytex: ,
FLEX: FLOAT yytex: 6.0 -> 6.000000
FLEX: COMMA yytex: ,
FLEX: TTRUE yytex: true
FLEX: COMMA yytex: ,
FLEX: STRING yytex: afds -> afds
FLEX: COMMA yytex: ,
FLEX: ID_B yytex: b -> b
BISON: Looking for ID b
FLEX: SQUARE_BRACKET_CLOSE yytex: ]
FLEX: NEWLINE
BISON: ASSIGN ID: c TYPE: ARRAY VALUE: 9, 4, 6.000000, true, afds, false
FLEX: ASSIGN yytex: :=
FLEX: SQUARE_BRACKET_OPEN yytex: [
FLEX: SQUARE_BRACKET_CLOSE yytex: ]
FLEX: NEWLINE
BISON: ASSIGN ID: d TYPE: ARRAY VALUE: true
FLEX: NEWLINE
FLEX: ID yytex: c -> c
FLEX: SQUARE_BRACKET_OPEN yytex: [
FLEX: INT yytex: 0 -> 0
FLEX: SQUARE_BRACKET_CLOSE yytex: ]
FLEX: NEWLINE
BISON: EXPRESSION: TYPE: INT VALUE: 9
FLEX: ID yytex: c -> c
FLEX: SQUARE_BRACKET_OPEN yytex: [
FLEX: INT yytex: 1 -> 1
FLEX: SQUARE_BRACKET_CLOSE yytex: ]
FLEX: NEWLINE
BISON: EXPRESSION: TYPE: INT VALUE: 4
FLEX: ID yytex: c -> c
FLEX: SQUARE_BRACKET_OPEN yytex: [
FLEX: INT yytex: 2 -> 2
FLEX: SQUARE_BRACKET_CLOSE yytex: ]
FLEX: NEWLINE
BISON: EXPRESSION: TYPE: FLOAT VALUE: 6.000000
The compilation has been successful
