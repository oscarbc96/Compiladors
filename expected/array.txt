EXPRESSION: TYPE: ARRAY VALUE: EMPTY
EXPRESSION: TYPE: ARRAY VALUE: 1
ASSIGN ID: a TYPE: INT VALUE: 9
ASSIGN ID: b TYPE: BOOL VALUE: false
ASSIGN ID: c TYPE: ARRAY VALUE: 9, 4, 6.000000, true, afds, false
ASSIGN ID: d TYPE: ARRAY VALUE: true
EXPRESSION: TYPE: INT VALUE: 9
EXPRESSION: TYPE: INT VALUE: 4
EXPRESSION: TYPE: FLOAT VALUE: 6.000000
